tests/Unit/IMAP/Charset/ConverterTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2023 Anna Larch <anna.larch@gmx.net>
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Tests\Unit\IMAP\Charset;
+
+use ChristophWurst\Nextcloud\Testing\TestCase;
+use Horde_Imap_Client;
+use Horde_Imap_Client_Data_Fetch;
+use Horde_Imap_Client_Fetch_Query;
+use Horde_Imap_Client_Fetch_Results;
+use Horde_Imap_Client_Ids;
+use Horde_Imap_Client_Socket;
+use Horde_Mime_Part;
+use OCA\Mail\Db\Mailbox;
+use OCA\Mail\IMAP\Charset\Converter;
+use OCA\Mail\IMAP\ImapMessageFetcher;
+use OCA\Mail\IMAP\ImapMessageFetcherFactory;
+use OCA\Mail\IMAP\MessageMapper;
+use OCA\Mail\Model\IMAPMessage;
+use OCA\Mail\Service\SmimeService;
+use OCA\Mail\Support\PerformanceLoggerTask;
+use PHPUnit\Framework\MockObject\MockObject;
+use Psr\Log\LoggerInterface;
+use function range;
+
+class ConverterTest extends TestCase {
+
+	public Converter $converter;
+	protected function setUp(): void {
+		parent::setUp();
+
+		$this->converter = new Converter();
+	}
+
+	/**
+	 * @dataProvider dataProviderMimeParts
+	 */
+	public function testConvert($mimePart, $expected): void {
+		$actual = $this->converter->convert($mimePart);
+		$this->assertEquals($expected, $actual);
+		$isUtf8 = mb_check_encoding($actual, 'UTF-8');
+		$this->assertTrue($isUtf8);
+	}
+
+	public function dataProviderMimeParts(): array {
+		// UTF8
+		$utfMimePart = new Horde_Mime_Part();
+		$utfMimePart->setType('text/plain');
+		$utfMimePart->setCharset('UTF-8');
+		$utfMimePart->setContents('😊');
+		// Hebrew
+		$iso88591MimePart = new Horde_Mime_Part();
+		$iso88591MimePart->setType('text/plain');
```

Comment:
the parts have to be text/* so horde even considers the charset when read back. default parts are application and have no charset